primitives/state-machine/src/lib.rs

```diff
@@ -55,11 +55,22 @@ pub use tracing::trace;
 #[cfg(not(feature = "std"))]
 #[macro_export]
 macro_rules! warn {
-	(target: $target:expr, $($arg:tt)+) => {
+	(target: $target:expr, $message:expr) => {
 		()
 	};
-	($($arg:tt)+) => {
-		()
+	(target: $target:expr, $message:expr, $( $arg:ident ),*) => {
+		{
+			$(
+				let _ = &$arg;
+			)*
+		}
+	};
+	($message:expr, $( $arg:expr, )*) => {
```

Comment:
shouldn't we have the commas here optional, the one after message and the last one, maybe we can write:
```suggestion
	($message:expr $( , $arg:expr )* $( , )?) => {
```
If it works we can use the same trick in the 2 match with `target` above and merge them into: `target: $target:expr, $message:expr $( , $arg:ident )* $( , )?`

EDIT: anyway everybody should use log instead.